src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -168,6 +170,7 @@ internal JsonTypeInfo(Type type, JsonConverter converter, JsonSerializerOptions
             Options = options;
             PropertyInfoForTypeInfo = CreatePropertyInfoForTypeInfo(Type, converter, Options, this);
             ElementType = converter.ElementType;
+            ConfigurePolymorphism(converter, options);
```

Comment:
This should happen in the Configure method

Reply:
treat as optional because neither converter nor options are/will be possible to change by user but I assume we will want to add some more configuration here in the future so we can as well delay that now

Reply:
I know, I will need to change this once polymorphism configuration becomes a user-settable property, but couldn't find a good reason to make the change right now.